pycqed/analysis_v2/base_analysis.py

```diff
@@ -372,7 +386,14 @@ def add_measured_data(raw_data_dict, compression_factor=1):
                     ssp = np.arange(len(ssp) * compression_factor)
                 raw_data_dict['hard_sweep_points'] = hsp
                 raw_data_dict['soft_sweep_points'] = ssp
-
+            elif sp is not None:
+                # deal with hybrid measurements
+                if sweep_points.shape[0] == 1 and len(sp) > 1:
+                    hybrid_measurement = True
+                    hsp = np.arange(len(next(iter(sp[0].values()))[0]))
```

Comment:
quite compact but a bit hard to read, I must say :D 5 nested function calls + 2 indexing... maybe doing it in 2 steps would make it more readable? or a comment explaining what happens (just imagine somehow the inner structure of SweepPoints changes and someone is asked to check if the code would work here with the new structure, would be pretty hard to say just by looking at this one liner :p)

Reply:
This is done in many places throughout timedomain_analysis :p And basically every time I need to access some parameters from SweepPoints. I don't want to have 2 lines of code every time I need some parameters from SweepPoints.
It's true that here I take np.arange and len in addition to the next(iter()) so I will split it in 2 lines.

Reply:
if you need it each time you have to access sweep points than the even nicer solution is to create a function within the sweeppoint object. This is best because also it hides inner structure of sweeppoints and in case the structure changes, we can just change that "getter" function. A bit like the "cal_points.get_states()"

Reply:
yes, that is a good solution. I will need to go through everywhere I did this but it's better to do it now than later after we will have used it in many more places :P

Reply:
I have done this now in an atomic commit with SHA-1: 92fcca176b94028c2470dbfd1ecb50bdfeb5bd10. We should cherry pick this commit into our paralle_2qb_calib_class branch so we use them there.